spec/subsequence-provider-spec.js

```diff
@@ -0,0 +1,557 @@
+          expect(suggestions[0].text).toBe('abcomment')
+          expect(suggestions[0].type).toBe('comment')
+          expect(suggestions[1].text).toBe('abcdef')
+          expect(suggestions[1].type).toBe('builtin')
+        })
+      )
+    })
+  })
+
+  describe('when the symbols config contains a list of suggestion objects', () => {
+    beforeEach(() => {
+      editor.setText('// abcomment')
+      waitForBufferToStopChanging()
+
+      let commentConfig = {
+        comment: { selector: '.comment' },
+        builtin: {
+          suggestions: [
+            {nope: 'nope1', rightLabel: 'will not be added to the suggestions'},
+            {text: 'abcd', rightLabel: 'one', type: 'function'},
+            []
+          ]
+        }
+      }
+      atom.config.set('autocomplete.symbols', commentConfig, {scopeSelector: '.source.js .comment'})
+    })
+
+    it('adds the suggestion objects to the results', () => {
+      // Using the comment config
+      editor.setCursorBufferPosition([0, 2])
+
+      waitsForPromise(() =>
+        suggestionsForPrefix(provider, editor, 'ab', {raw: true}).then(suggestions => {
+          expect(suggestions).toHaveLength(2)
+          expect(suggestions[0].text).toBe('abcomment')
+          expect(suggestions[0].type).toBe('comment')
+          expect(suggestions[1].text).toBe('abcd')
+          expect(suggestions[1].type).toBe('function')
+          expect(suggestions[1].rightLabel).toBe('one')
+        })
+      )
+    })
+  })
+
+  describe('when the legacy completions array is used', () => {
+    beforeEach(() => {
+      editor.setText('// abcomment')
+      waitForBufferToStopChanging()
+      atom.config.set('editor.completions', ['abcd', 'abcde', 'abcdef'], {scopeSelector: '.source.js .comment'})
+    })
+
+    it('uses the config for the scope under the cursor', () => {
+      // Using the comment config
+      editor.setCursorBufferPosition([0, 2])
+
+      waitsForPromise(() =>
+        suggestionsForPrefix(provider, editor, 'ab', {raw: true}).then(suggestions => {
+          expect(suggestions).toHaveLength(4)
+          expect(suggestions[0].text).toBe('abcomment')
+          expect(suggestions[0].type).toBe('')
+          expect(suggestions[1].text).toBe('abcdef')
+          expect(suggestions[1].type).toBe('builtin')
+        })
+      )
+    })
+  })
+
+  // TODO: support unicode
+  // it('adds words to the wordlist with unicode characters', done => {
+  //   atom.config.set('autocomplete-plus.enableExtendedUnicodeSupport', true)
+  //   suggestionsForPrefix(provider, editor, 'somē', {raw: true}).then(suggestions => {
+  //     expect(suggestions).toHaveLength(0)
+  //     editor.insertText('somēthingNew')
+  //     editor.insertText(' ')
+  //     waitForBufferToStopChanging()
+  //
+  //     return suggestionsForPrefix(provider, editor, 'somē', {raw: true})
+  //   }).then(suggestions => {
+  //     expect(suggestions).toHaveLength(1)
+  //     done()
+  //   })
+  // })
```

Comment:
Do you have a sense as to why this doesn't work today? Does `std::iswalpha` returns false for the `ē` character?